src/backend/InvenTree/report/api.py

```diff
@@ -1,164 +1,314 @@
 
-                    try:
-                        if not item_query.filter(**filters).exists():
-                            matches = False
-                            break
-                    except FieldError:
-                        matches = False
-                        break
+    class Meta:
+        """Filter options."""
 
-                # Matched all items
-                if matches:
-                    valid_report_ids.add(report.pk)
+        model = report.models.ReportTemplate
+        fields = ['landscape']
 
-            # Reduce queryset to only valid matches
-            queryset = queryset.filter(pk__in=list(valid_report_ids))
 
-        return queryset
+class LabelFilter(ReportFilterBase):
+    """Filter class for label template list."""
 
+    class Meta:
+        """Filter options."""
 
-@method_decorator(cache_page(5), name='dispatch')
-class ReportPrintMixin:
-    """Mixin for printing reports."""
+        model = report.models.LabelTemplate
+        fields = []
+
+
+class LabelPrint(GenericAPIView):
+    """API endpoint for printing labels."""
+
+    permission_classes = [permissions.IsAuthenticated]
+    serializer_class = report.serializers.LabelPrintSerializer
+
+    def get_plugin_class(self, plugin_id: int, raise_error=False):
+        """Return the plugin class for the given plugin key."""
+        from plugin.models import PluginConfig
+
+        plugin = None
+
+        try:
+            plugin_config = PluginConfig.objects.get(pk=plugin_id)
+            plugin = plugin_config.plugin
+        except (ValueError, PluginConfig.DoesNotExist):
+            pass
+
+        error = None
+
+        if not plugin:
+            error = _('Plugin not found')
+        elif not plugin.is_active():
+            error = _('Plugin is not active')
+        elif not plugin.mixin_enabled('labels'):
+            error = _('Plugin does not support label printing')
+
+        if error:
+            plugin = None
+
+            if raise_error:
+                raise ValidationError({'plugin': error})
+
+        return plugin
+
+    def get_plugin_serializer(self, plugin):
+        """Return the serializer for the given plugin."""
+        if plugin and hasattr(plugin, 'get_printing_options_serializer'):
+            return plugin.get_printing_options_serializer(self.request)
+
+        return None
+
+    def get_serializer(self, *args, **kwargs):
+        """Return serializer information for the label print endpoint."""
+        plugin = None
+
+        # Plugin information provided?
+        if self.request:
+            plugin_key = self.request.query_params.get('plugin', None)
+            plugin = self.get_plugin_class(plugin_key)
+            plugin_serializer = self.get_plugin_serializer(plugin)
+
+            if plugin_serializer:
+                kwargs['plugin_serializer'] = plugin_serializer
+
+        serializer = super().get_serializer(*args, **kwargs)
+        return serializer
```

Comment:
@SchrodingersGat are you sure this is working? I can not get PrintingOptionsSerializer  running in OPTIONS calls.

Reply:
I'll check again when I am back at my computer but yeah it has been working

Reply:
The OPTIONS calls too? I have a very hard time getting the tests running consistently in this regard

Reply:
I have found the issue, you can not reload/activate/deactivate plugins between determining the plugin pk and doing any call using it as it seems to change